### ISSUE TITLE
Block variation: templateLock not working

### ISSUE THREAD

**Issue by huubl** [WordPress/gutenberg#64480]
### Description

Setting the templateLock attribute to contentOnly in a block variation is not working.

### Step-by-step reproduction instructions

1. Register the "cover-locked" block variation using the provided code.
2. Insert the "cover-locked" block into a post or page.
3. Save the post or page.
4. Reload the page.

### Screenshots, screen recording, code snippet

```
wp.blocks.registerBlockVariation("core/cover", {
	name: "cover-locked",
	title: "Cover Block Variation Locked",
	description: "When reloading the page, the templateLock 'contentOnly' is lost.",
	category: "media",
	icon: "smiley",
	attributes: {
		templateLock: "contentOnly",
		useFeaturedImage: true,
		dimRatio: 0,
		backgroundType: "image",
		overlayColor: "",
		isUserOverlayColor: true,
		contentPosition: "center center",
		isDark: false,
		className: "is-cover-locked",
		tagName: "div",
		textColor: "white",
		layout: {
			type: "constrained",
			contentSize: ""
		},
	},
	innerBlocks: [
		["core/heading", {level: 1, placeholder: "When reloading the page, the templateLock 'contentOnly' is lost."}],
	],
	example: undefined,
	scope: ["inserter"],
	isDefault: false,
	isActive: ['className'],
});
```

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [X] Yes

**Comment by talldan** [comment]
Thanks for reporting, I could reproduce it. It looks to be the same kind of issue we had for the group block recently (see - https://github.com/WordPress/gutenberg/pull/63837).

There's a block deprecation that causes the attributes to be stripped:
https://github.com/WordPress/gutenberg/blob/46f9fe0f31bf9ce5f6210cad96074206e90527e6/packages/block-library/src/cover/deprecated.js#L401-L412

The `isElligible` function is the main issue. `customOverlayColor` or `overlayColor` are currently in use and valid block attributes for the cover block, so I don't understand why this implementation checks for them. It causes the deprecation to match against perfectly normal use cases of the block. 

The issue was introduced in https://github.com/WordPress/gutenberg/pull/54054 (cc @ajlende)

**Comment by andrewserong** [comment]
I think I have a fix for this over in #65105.
